### ISSUE TITLE
XSD1.1 duplicate identity constraint caused by xs:redefine

### ISSUE THREAD

**Issue by Malertach** [sissaschool/xmlschema#259]
When redefining a xs:complexType, the xs:unique constraint of its xs:element seems to get doubled. The creation of the schema instance fails with the following traceback:
```python
Python 3.7.0 (v3.7.0:1bf9cc5093, Jun 27 2018, 04:06:47) [MSC v.1914 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import xmlschema
>>> my_schema = xmlschema.XMLSchema11('Myschema2.xsd')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\Malertach\AppData\Local\Programs\Python\Python37-32\lib\site-packages\xmlschema\validators\schema.py", line 477, in __init__
    self.maps.build()
  File "C:\Users\Malertach\AppData\Local\Programs\Python\Python37-32\lib\site-packages\xmlschema\validators\global_maps.py", line 608, in build
    group.build()
  File "C:\Users\Malertach\AppData\Local\Programs\Python\Python37-32\lib\site-packages\xmlschema\validators\groups.py", line 249, in build
    self._group[k] = element_class(elem, schema, self)
  File "C:\Users\Malertach\AppData\Local\Programs\Python\Python37-32\lib\site-packages\xmlschema\validators\xsdbase.py", line 234, in __init__
    self.elem = elem
  File "C:\Users\Malertach\AppData\Local\Programs\Python\Python37-32\lib\site-packages\xmlschema\validators\elements.py", line 99, in __setattr__
    super(XsdElement, self).__setattr__(name, value)
  File "C:\Users\Malertach\AppData\Local\Programs\Python\Python37-32\lib\site-packages\xmlschema\validators\xsdbase.py", line 246, in __setattr__
    self._parse()
  File "C:\Users\Malertach\AppData\Local\Programs\Python\Python37-32\lib\site-packages\xmlschema\validators\elements.py", line 1108, in _parse
    self._parse_constraints()
  File "C:\Users\Malertach\AppData\Local\Programs\Python\Python37-32\lib\site-packages\xmlschema\validators\elements.py", line 274, in _parse_constraints
    self.parse_error("duplicated identity constraint %r:" % constraint.name, child)
  File "C:\Users\Malertach\AppData\Local\Programs\Python\Python37-32\lib\site-packages\xmlschema\validators\xsdbase.py", line 172, in parse_error
    raise error
xmlschema.validators.exceptions.XMLSchemaParseError: duplicated identity constraint 'redChannelNameUniqueness':

Schema:

  <xs:unique xmlns:xs="http://www.w3.org/2001/XMLSchema" name="redChannelNameUniqueness">
    <xs:selector xpath="redChannel" />
    <xs:field xpath="@name" />
  </xs:unique>

Path: /xs:schema/xs:complexType[1]/xs:all/xs:element[4]/xs:unique

Schema URL: file:///C:/Users/Malertach/Desktop/xmlschema/Myschema1.xsd

Origin URL: file:///C:/Users/Malertach/Desktop/xmlschema/Myschema2.xsd

>>>
```
The used schemas are attached:
[xmlschema.zip](https://github.com/sissaschool/xmlschema/files/7066087/xmlschema.zip)

**Comment by brunato** [comment]
Hi,

is not a real duplication but a wrong check in `XsdElement._parse_constraints()`:

https://github.com/sissaschool/xmlschema/blob/d5808de27db09f5060adc6da8850abde561a1d38/xmlschema/validators/elements.py#L272-L278

the comparison at line 273 has to be changed to:

```python
child != self.maps.identities[constraint.name].elem
```

Thank you for reporting this, the fix will be available soon in a new release.

**Comment by brunato** [comment]
Fix available with the release v1.7.1.
